framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/OpenShiftDeployment.java

```diff
@@ -144,30 +147,68 @@ protected void waitUntilAllPodsAreReadyAndRunning(int expectedPods) {
 
     @Override
     public void setRouterTimeout(Duration timeoutValue) {
-        // Route has a same name as its service.
-        String routeName = getServiceName();
-        util.client()
-            .routes()
-            .withName(routeName)
-            .edit()
-            .editMetadata()
-            .addToAnnotations(OpenShiftConstants.HAPROXY_ROUTER_TIMEOUT, timeoutValue.getSeconds()+"s")
-            .endMetadata()
-            .done();
+        RouteList list = getRoutes();
+        for(Route r : list.getItems()) {
+          util.client()
+              .routes()
+              .withName(r.getMetadata().getName())
+              .edit()
+              .editMetadata()
+              .addToAnnotations(OpenShiftConstants.HAPROXY_ROUTER_TIMEOUT, timeoutValue.getSeconds()+"s")
+              .endMetadata()
+              .done();
+        }
     }
 
     @Override
     public void resetRouterTimeout() {
-        // Route has a same name as its service.
-        String routeName = getServiceName();
+        RouteList list = getRoutes();
+        for(Route r : list.getItems()) {
+          util.client()
+              .routes()
+              .withName(r.getMetadata().getName())
+              .edit()
+              .editMetadata()
+              .removeFromAnnotations(OpenShiftConstants.HAPROXY_ROUTER_TIMEOUT)
+              .endMetadata()
+              .done();
+        }
+    }
+    
+    @Override
+    public void setRouterBalance(String balance) {
+        RouteList list = getRoutes();
+        for(Route r : list.getItems()) {
+          util.client()
+              .routes()
+              .withName(r.getMetadata().getName())
+              .edit()
+              .editMetadata()
+              .addToAnnotations(OpenShiftConstants.HAPROXY_ROUTER_BALANCE, balance)
+              .endMetadata()
+              .done();
+        }
+    }
+    
+    @Override
+    public void setResources(Map<String,String> requests, Map<String,String> limits) {
         util.client()
-            .routes()
-            .withName(routeName)
+            .deploymentConfigs()
+            .withName(getServiceName())
```

Comment:
this can be confusing to read. Can you put there a comment
```suggestion
            .withName(getServiceName()) // Deployment config has a same name as its service.
```

Reply:
Done